src/main/java/io/reactivex/Single.java

```diff
@@ -1588,7 +1588,7 @@
      */
     public final Single<T> doOnSuccess(final Consumer<? super T> onSuccess) {
         ObjectHelper.requireNonNull(onSuccess, "onSuccess is null");
-        return RxJavaPlugins.onAssembly(new SingleDoOnSuccess<T>(this, onSuccess));
+        return RxJavaPlugins.onAssembly(new SingleDoOnEvent<T>(this, onSuccess, Functions.emptyConsumer()));
```

Comment:
True I just thought of reusing it since that's also the way it is done in Completable